dist_utils.py

```diff
@@ -38,9 +38,10 @@ def wheel_name(pkg_name, version, python_version, platform_tag):
 
 def get_version_from_source_tree(source_tree):
     version_file_path = '{}/cupy/_version.py'.format(source_tree)
+    exec_locals = {}
     with open(version_file_path) as f:
-        exec(f.read())
-    return locals()['__version__']
+        exec(f.read(), None, exec_locals)
+    return exec_locals['__version__']
```

Comment:
I guess that this fix is intended to prevent `locals()` from begin mutated, but how is it related to this pull request?

Reply:
It's to adopt Python 3.13's change about `locals()` within optimized scopes (e.g. functions):
https://docs.python.org/3.13/whatsnew/3.13.html#defined-mutation-semantics-for-locals